lib/bluepill/controller.rb

```diff
@@ -86,6 +86,7 @@ def grep_pattern(application, query = nil)
     def cleanup_bluepill_directory
       running_applications.each do |app|
         pid = pid_for(app)
+        next unless pid
         next if pid || System.pid_alive?(pid)
```

Comment:
Yup - I read the code and looking at the prior commit / line - https://github.com/bluepill-rb/bluepill/commit/e4b759b9a78f65a8e339b5282adbf29d52b77880#diff-1bc4b493e767946e9043d0c7cd3ec3ebL90

That "next if" condition logic changes - In the version prior to that commit, the inside conditional block starting at line 90 skips only if both pid and System.pid_alive are true (```if !pid || !System.pid_alive?(pid)``` evaluates to false only if both pid and System.pid_alive each evaluate to true, eg, it enters the block)... After that commit, it skips that block if either are true.

The proper fix is to change line 89 to ```next if pid && System.pid_alive?(pid)``` and delete the line I added.